lerna run fails on colon-separated scripts with lerna+Nx

## Current Behavior

If a module package script has a colon in the name (for instance "this:example") then lerna+Nx run cannot run it, instead failing with a pair of errors below:
```
> nx-npm-script-name-bug@1.0.0 test /dev/nx-task-name-bug
> lerna run script:with:colon

lerna notice cli v5.1.6
Error: Cannot find configuration for task @mybug/module-with-colon-script:script
    at getExecutorNameForTask (/dev/nx-task-name-bug/node_modules/nx/src/tasks-runner/utils.js:123:15)
    at getExecutorForTask (/dev/nx-task-name-bug/node_modules/nx/src/tasks-runner/utils.js:129:22)
    at getCustomHasher (/dev/nx-task-name-bug/node_modules/nx/src/tasks-runner/utils.js:136:25)
    at TasksSchedule.<anonymous> (/dev/nx-task-name-bug/node_modules/nx/src/tasks-runner/tasks-schedule.js:112:62)
    at Generator.next (<anonymous>)
    at /dev/nx-task-name-bug/node_modules/tslib/tslib.js:118:75
    at new Promise (<anonymous>)
    at Object.__awaiter (/dev/nx-task-name-bug/node_modules/tslib/tslib.js:114:16)
    at TasksSchedule.hashTask (/dev/nx-task-name-bug/node_modules/nx/src/tasks-runner/tasks-schedule.js:111:24)
    at TasksSchedule.<anonymous> (/dev/nx-task-name-bug/node_modules/nx/src/tasks-runner/tasks-schedule.js:59:24)
Unexpected error:
Error: Unable to load hasher for task "@mybug/module-with-colon-script:script"
    at getCustomHasher (/dev/nx-task-name-bug/node_modules/nx/src/tasks-runner/utils.js:141:15)
    at TasksSchedule.<anonymous> (/dev/nx-task-name-bug/node_modules/nx/src/tasks-runner/tasks-schedule.js:112:62)
    at Generator.next (<anonymous>)
    at /dev/nx-task-name-bug/node_modules/tslib/tslib.js:118:75
    at new Promise (<anonymous>)
    at Object.__awaiter (/dev/nx-task-name-bug/node_modules/tslib/tslib.js:114:16)
    at TasksSchedule.hashTask (/dev/nx-task-name-bug/node_modules/nx/src/tasks-runner/tasks-schedule.js:111:24)
    at TasksSchedule.<anonymous> (/dev/nx-task-name-bug/node_modules/nx/src/tasks-runner/tasks-schedule.js:59:24)
    at Generator.next (<anonymous>)
    at /dev/nx-task-name-bug/node_modules/tslib/tslib.js:118:75
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! nx-npm-script-name-bug@1.0.0 test: `lerna run script:with:colon`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the nx-npm-script-name-bug@1.0.0 test script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /.npm/_logs/2022-06-30T16_10_59_095Z-debug.log
```

## Expected Behavior

The script should execute, as occurs when `useNx: false` is set in the lerna.json config file:
```
> nx-npm-script-name-bug@1.0.0 test /dev/nx-task-name-bug
> lerna run script:with:colon

lerna notice cli v5.1.6
lerna info Executing command in 1 package: "npm run script:with:colon"
lerna info run Ran npm script 'script:with:colon' in '@mybug/module-with-colon-script' in 0.3s:

> @mybug/module-with-colon-script@1.0.0 script:with:colon /dev/nx-task-name-bug/packages/module-with-colon-script
> echo This Works!

This Works!
lerna success run Ran npm script 'script:with:colon' in 1 package in 0.3s:
lerna success - @mybug/module-with-colon-script
```

## Steps to Reproduce

Covered here: [Steps to reproduce](https://github.com/brees-worth/lerna-script-bug)

### Failure Logs / Configuration

<details><summary>lerna.json</summary><p>

```json
{
  "packages": [
    "packages/*"
  ],
  "useNx": true,
  "version": "0.0.0"
}
```

</p></details>

### Environment

```
lerna notice cli v5.1.6

 Environment info:

  System:
    OS: macOS 12.4
    CPU: (8) x64 Intel(R) Core(TM) i5-8259U CPU @ 2.30GHz
  Binaries:
    Node: 14.19.1 - ~/.nvm/versions/node/v14.19.1/bin/node
    Yarn: 1.22.18 - /usr/local/bin/yarn
    npm: 6.14.17 - ~/.nvm/versions/node/v14.19.1/bin/npm
  Utilities:
    Git: 2.32.1 - /usr/bin/git
  npmPackages:
    lerna: ^5.1.6 => 5.1.6 
```

--- a/commands/run/__tests__/__fixtures__/powered-by-nx/packages/package-1/package.json
+++ b/commands/run/__tests__/__fixtures__/powered-by-nx/packages/package-1/package.json
@@ -3,6 +3,7 @@
   "version": "1.0.0",
   "scripts": {
     "fail": "exit 1",
-    "my-script": "echo package-1"
+    "my-script": "echo package-1",
+    "another-script:but-with-colons": "echo package-1-script-with-colons"
   }
 }

--- a/commands/run/__tests__/run-command.test.js
+++ b/commands/run/__tests__/run-command.test.js
@@ -331,6 +331,13 @@ describe("RunCommand", () => {
       expect(collectedOutput).toContain("Successfully ran target");
     });
 
+    it("runs a script with a colon in the script name", async () => {
+      collectedOutput = "";
+      await lernaRun(testDir)("another-script:but-with-colons");
+      expect(collectedOutput).toContain("package-1-script-with-colons");
+      expect(collectedOutput).toContain("Successfully ran target");
+    });
+
     it("runs a script only in scoped packages", async () => {
       collectedOutput = "";
       await lernaRun(testDir)("my-script", "--scope", "package-1");

--- a/commands/run/index.js
+++ b/commands/run/index.js
@@ -179,7 +179,7 @@ class RunCommand extends Command {
     const { targetDependencies, options } = this.prepNxOptions();
     if (this.packagesWithScript.length === 1) {
       const { runOne } = require("nx/src/command-line/run-one");
-      const fullQualifiedTarget = this.packagesWithScript.map((p) => p.name)[0] + ":\"" + this.script+ "\"";
+      const fullQualifiedTarget = this.packagesWithScript.map((p) => p.name)[0] + ':"' + this.script + '"';
       return runOne(
         process.cwd(),
         {

--- a/commands/run/index.js
+++ b/commands/run/index.js
@@ -172,7 +172,7 @@ class RunCommand extends Command {
 
   addQuotesAroundScriptNameIfItHasAColon(scriptName) {
     // Nx requires quotes around script names of the form script:name
-    if(scriptName.includes(':')) {
+    if (scriptName.includes(":")) {
       return `"${scriptName}"`;
     } else {
       return scriptName;
@@ -188,7 +188,10 @@ class RunCommand extends Command {
     const { targetDependencies, options } = this.prepNxOptions();
     if (this.packagesWithScript.length === 1) {
       const { runOne } = require("nx/src/command-line/run-one");
-      const fullQualifiedTarget = this.packagesWithScript.map((p) => p.name)[0] + ':' + this.addQuotesAroundScriptNameIfItHasAColon(this.script);
+      const fullQualifiedTarget =
+        this.packagesWithScript.map((p) => p.name)[0] +
+        ":" +
+        this.addQuotesAroundScriptNameIfItHasAColon(this.script);
       return runOne(
         process.cwd(),
         {

--- a/commands/run/index.js
+++ b/commands/run/index.js
@@ -170,6 +170,15 @@ class RunCommand extends Command {
     return chain;
   }
 
+  addQuotesAroundScriptNameIfItHasAColon(scriptName) {
+    // Nx requires quotes around script names of the form script:name
+    if(scriptName.includes(':')) {
+      return `"${scriptName}"`;
+    } else {
+      return scriptName;
+    }
+  }
+
   runScriptsUsingNx() {
     if (this.options.ci) {
       process.env.CI = "true";
@@ -179,7 +188,7 @@ class RunCommand extends Command {
     const { targetDependencies, options } = this.prepNxOptions();
     if (this.packagesWithScript.length === 1) {
       const { runOne } = require("nx/src/command-line/run-one");
-      const fullQualifiedTarget = this.packagesWithScript.map((p) => p.name)[0] + ':"' + this.script + '"';
+      const fullQualifiedTarget = this.packagesWithScript.map((p) => p.name)[0] + ':' + this.addQuotesAroundScriptNameIfItHasAColon(this.script);
       return runOne(
         process.cwd(),
         {

--- a/commands/run/index.js
+++ b/commands/run/index.js
@@ -179,7 +179,7 @@ class RunCommand extends Command {
     const { targetDependencies, options } = this.prepNxOptions();
     if (this.packagesWithScript.length === 1) {
       const { runOne } = require("nx/src/command-line/run-one");
-      const fullQualifiedTarget = this.packagesWithScript.map((p) => p.name)[0] + ":" + this.script;
+      const fullQualifiedTarget = this.packagesWithScript.map((p) => p.name)[0] + ":\"" + this.script+ "\"";
       return runOne(
         process.cwd(),
         {

--- a/e2e/tests/lerna-run/lerna-run.spec.ts
+++ b/e2e/tests/lerna-run/lerna-run.spec.ts
@@ -317,6 +317,7 @@ describe("useNx", () => {
       packagePath: "packages/package-4",
       scripts: {
         "print:name": "echo test-package-4",
+        "print:name:run-one-only": "echo test-package-4-run-one-only",
       },
     });
     await fixture.lerna("create package-5 -y");
@@ -369,6 +370,35 @@ test-package-X
  
 
  >  Lerna (powered by Nx)   Successfully ran target print-name for 3 projects
+
+
+lerna notice cli v999.9.9-e2e.0
+
+`);
+  });
+
+  it("should run script with colon on single child package using nx", async () => {
+    const output = await fixture.lerna(`run print:name:run-one-only`);
+
+    expect(output.combinedOutput).toMatchInlineSnapshot(`
+
+ >  Lerna (powered by Nx)   Running target print:name:run-one-only for 1 project(s):
+
+    - package-X
+
+ 
+
+> package-X:"print:name:run-one-only"
+
+
+> package-X@0.0.0 print:name:run-one-only
+> echo test-package-X-run-one-only
+
+test-package-X-run-one-only
+
+ 
+
+ >  Lerna (powered by Nx)   Successfully ran target print:name for 1 projects
 
 
 lerna notice cli v999.9.9-e2e.0

--- a/e2e/tests/lerna-run/lerna-run.spec.ts
+++ b/e2e/tests/lerna-run/lerna-run.spec.ts
@@ -378,10 +378,9 @@ lerna notice cli v999.9.9-e2e.0
 `);
   });
 
-  describe('run one', () => {
+  describe("run one", () => {
     it("should run script on single child package using nx", async () => {
       const output = await fixture.lerna(`run print-name-run-one-only`);
-      
 
       expect(output.combinedOutput).toMatchInlineSnapshot(`
   
@@ -402,10 +401,10 @@ lerna notice cli v999.9.9-e2e.0
   
   `);
     });
-  
+
     it("should run script with colon on single child package using nx", async () => {
       const output = await fixture.lerna(`run print:name:run-one-only`);
-  
+
       expect(output.combinedOutput).toMatchInlineSnapshot(`
 
   > package-X:"print:name:run-one-only"
@@ -425,7 +424,7 @@ lerna notice cli v999.9.9-e2e.0
   
   `);
     });
-  })
+  });
 
   it("should run script with colon on all child package using nx", async () => {
     const output = await fixture.lerna(`run print:name`);

--- a/e2e/tests/lerna-run/lerna-run.spec.ts
+++ b/e2e/tests/lerna-run/lerna-run.spec.ts
@@ -312,6 +312,20 @@ describe("useNx", () => {
         "print-name": "echo test-package-3",
       },
     });
+    await fixture.lerna("create package-4 -y");
+    await fixture.addScriptsToPackage({
+      packagePath: "packages/package-4",
+      scripts: {
+        "print:name": "echo test-package-4",
+      },
+    });
+    await fixture.lerna("create package-5 -y");
+    await fixture.addScriptsToPackage({
+      packagePath: "packages/package-5",
+      scripts: {
+        "print:name": "echo test-package-5",
+      },
+    });
   });
   afterAll(() => fixture.destroy());
 
@@ -355,6 +369,44 @@ test-package-X
  
 
  >  Lerna (powered by Nx)   Successfully ran target print-name for 3 projects
+
+
+lerna notice cli v999.9.9-e2e.0
+
+`);
+  });
+
+  it("should run script with colon on all child package using nx", async () => {
+    const output = await fixture.lerna(`run print:name`);
+
+    expect(output.combinedOutput).toMatchInlineSnapshot(`
+
+ >  Lerna (powered by Nx)   Running target print:name for 2 project(s):
+
+    - package-X
+    - package-X
+
+ 
+
+> package-X:"print:name"
+
+
+> package-X@0.0.0 print:name
+> echo test-package-X
+
+test-package-X
+
+> package-X:"print:name"
+
+
+> package-X@0.0.0 print:name
+> echo test-package-X
+
+test-package-X
+
+ 
+
+ >  Lerna (powered by Nx)   Successfully ran target print:name for 2 projects
 
 
 lerna notice cli v999.9.9-e2e.0

--- a/e2e/tests/lerna-run/lerna-run.spec.ts
+++ b/e2e/tests/lerna-run/lerna-run.spec.ts
@@ -317,7 +317,8 @@ describe("useNx", () => {
       packagePath: "packages/package-4",
       scripts: {
         "print:name": "echo test-package-4",
-        "print:name:run-one-only": "echo test-package-4-run-one-only",
+        "print-name-run-one-only": "echo test-package-4-run-one-only",
+        "print:name:run-one-only": "echo test-package-4-run-one-only-with-colon",
       },
     });
     await fixture.lerna("create package-5 -y");
@@ -377,34 +378,54 @@ lerna notice cli v999.9.9-e2e.0
 `);
   });
 
-  it("should run script with colon on single child package using nx", async () => {
-    const output = await fixture.lerna(`run print:name:run-one-only`);
+  describe('run one', () => {
+    it("should run script on single child package using nx", async () => {
+      const output = await fixture.lerna(`run print-name-run-one-only`);
+      
 
-    expect(output.combinedOutput).toMatchInlineSnapshot(`
+      expect(output.combinedOutput).toMatchInlineSnapshot(`
+  
+  > package-X:print-name-run-one-only
+  
+  
+  > package-X@0.0.0 print-name-run-one-only
+  > echo test-package-X-run-one-only
+  
+  test-package-X-run-one-only
+  
+   
+  
+   >  Lerna (powered by Nx)   Successfully ran target print-name-run-one-only for project package-X
+  
+  
+  lerna notice cli v999.9.9-e2e.0
+  
+  `);
+    });
+  
+    it("should run script with colon on single child package using nx", async () => {
+      const output = await fixture.lerna(`run print:name:run-one-only`);
+  
+      expect(output.combinedOutput).toMatchInlineSnapshot(`
 
- >  Lerna (powered by Nx)   Running target print:name:run-one-only for 1 project(s):
-
-    - package-X
-
- 
-
-> package-X:"print:name:run-one-only"
-
-
-> package-X@0.0.0 print:name:run-one-only
-> echo test-package-X-run-one-only
-
-test-package-X-run-one-only
-
- 
-
- >  Lerna (powered by Nx)   Successfully ran target print:name for 1 projects
-
-
-lerna notice cli v999.9.9-e2e.0
-
-`);
-  });
+  > package-X:"print:name:run-one-only"
+  
+  
+  > package-X@0.0.0 print:name:run-one-only
+  > echo test-package-X-run-one-only-with-colon
+  
+  test-package-X-run-one-only-with-colon
+  
+   
+  
+   >  Lerna (powered by Nx)   Successfully ran target print:name:run-one-only for project package-X
+  
+  
+  lerna notice cli v999.9.9-e2e.0
+  
+  `);
+    });
+  })
 
   it("should run script with colon on all child package using nx", async () => {
     const output = await fixture.lerna(`run print:name`);

